.github/workflows/container-pr.yml

```diff
@@ -12,44 +12,59 @@ concurrency:
   group: ${{ github.head_ref }}-container
   cancel-in-progress: true
 jobs:
-  validate:
+  bump_version:
     runs-on: ubuntu-latest
-    if: ${{ github.actor != 'dependabot[bot]' }}
-
+    permissions:
+      contents: write
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      - name: Set up Google Cloud Credentials
-        uses: google-github-actions/auth@v1
-        with:
-          credentials_json: ${{ secrets.GOOGLE_HOPRASSOCIATION_CREDENTIALS_REGISTRY }}
+      - name: Bump Cargo version
+        id: bumping_version
+        run: |
+          hopr_operator_version=$(grep "^version = " Cargo.toml | sed 's/.* "//' | sed 's/"$//' | tr -d '\n')
+          git_tag=$(git tag -l ${hopr_operator_version})
+          echo "Workspace version is ${hopr_operator_version} while tagged is ${git_tag}"
+          if [ "${git_tag}" = "${hopr_operator_version}" ]; then
+            patch_number=$(echo $git_tag | sed 's/.*\.//')
+            release_number=$(echo $git_tag | sed "s/\.${patch_number}//")
+            bump_version=${release_number}.$((patch_number + 1))
```

Comment:
But how do you know that it is a patch update? It could be a minor update change... that would break semantic versioning. It's better to enforce that each PR that changes the Cargo files or src contents must change the version. And then in the target branch take that version and create a tag.

Reply:
E.g. on new PR open from dependabot... the version is automatically bumped (patch)... and it is up to the merge to make sure that the version is updated correctly.

Reply:
The idea is that if the PR owner does not change anything, then the workflow will force to update it, as the tag and the workspace version is the same. But if the PR owner modifies the workspace version by bumping minor or major, then, the workflow won't find the associated tag and won't do anyhting and will keep the new version provided.
It's kind of rudimentary but the solutions out there didn't work for the kind of release strategy that we have in this repository to create a release on every PR merge.